### ISSUE TITLE
Can't build on Windows / MSYS2 : error generating icons

### ISSUE THREAD

**Issue by rodolforg** [synfig/synfig#2817]
When I follow the steps, from scratch, describe here https://synfig-docs-dev.readthedocs.io/en/latest/building/Preparing%20Environment.html, I got this failure message and can't build Synfig Studio.

![image](https://user-images.githubusercontent.com/1323987/188320058-5a45a425-a9a5-4551-bac2-ef58d77d0edc.png)

I downloaded and installed the up-to-date MSYS2, in a completely new folder and still can't build it.

**Comment by rodolforg** [comment]
This build error happened when trying to build with CMake.

My bash_history:
```
pacman -Syu
pacman -S git
git clone https://github.com/synfig/synfig.git ~/synfig.git
cd synfig.git/
./1-setup-windows-msys2.sh
./2-build-msys-cmake.sh
```
```
$ cmake --version
cmake version 3.24.1
```
[CMakeCache.txt](https://github.com/synfig/synfig/files/9488796/CMakeCache.txt)

**Comment by ice0** [comment]
@rodolforg 

Can you run it again and show the full log?

**Comment by rodolforg** [comment]
OK.
I didn't reinstalled the setup this time, I cloned synfig git repository into another folder.

Here you are:
[log_step_1_b.txt](https://github.com/synfig/synfig/files/9489032/log_step_1_b.txt)
[log_step_1_b_err.txt](https://github.com/synfig/synfig/files/9489033/log_step_1_b_err.txt)
[log_step_2_b.txt](https://github.com/synfig/synfig/files/9489039/log_step_2_b.txt)
[log_step_2_b_err.txt](https://github.com/synfig/synfig/files/9489040/log_step_2_b_err.txt)

**Comment by ice0** [comment]
ok, got it. This happens because username contains non-Latin characters.

I thought we fixed all these issues, so I need time to check if it's a problem with synfig or cmake.

**Comment by ice0** [comment]
1. The first issue was incorrect handling of non-latin paths in the synfig CLI. Fixed by #2819

2. After that, `mod_magickpp` started to fail with the error:
```
exception 'Magick: NoEncodeDelegateForThisImageFormat `MIFF' @ error/constitute.c/WriteImage/1400'
```
I didn't debug why this is happening, so I just disabled it.
`cmake -DWITHOUT_MAGICPP=ON ...`

3. After that synfig fails to build .ico file. Fixed by #2820
4. After that CMake can't build resource file to embed icon to exe. Fixed by #2821

After all of this will be merged you still need to add `-DWITHOUT_MAGICPP=ON` to `2-build-msys-cmake.sh` so it will looks like:
```
...
cmake -DWITHOUT_MAGICPP=ON -DCMAKE_CXX_COMPILER_LAUNCHER=ccache -DCMAKE_C_COMPILER_LAUNCHER=ccache -DCMAKE_INSTALL_PREFIX=./install -DCMAKE_BUILD_TYPE=${mode} ..
cmake --build .
cmake --install . >/dev/null
```
